src/draive/scope/metrics.py

```diff
@@ -375,20 +376,32 @@ async def _summary(self) -> str:
 
         return summary
 
-    # TODO: combine all combinable metrics this way
-    # Warning: this method is not using lock
-    def _total_tokens_usage(self) -> TokenUsage:
-        total_usage: TokenUsage = TokenUsage()
-        match self._metrics.get(TokenUsage):
-            case TokenUsage() as usage:
-                total_usage.combine_metric(usage)
+    async def _combine_metric(self, metric: type[CombinableScopeMetric]) -> CombinableScopeMetric:
+        combined_metric = metric()
```

Comment:
metrics are not required to have empty initializers, we should not make them this way, if the metric was not defined it should remain None